### PR TITLE
CLDR-14571 json: improve bcp47

### PR DESCRIPTION
**Pull request by srl295**
- example JSON paths
  keyword.t.io.handwrit._description
  keyword.u.ms.metric._since

CLDR-14571

- [ ] This PR completes the ticket.

### CODE REVIEW

**Comment by srl295** [comment]
Examples:

**measure.json**
```json
{
  "keyword": {
    "u": {
      "ms": {
        "_deprecated": "false",
        "_description": "Measurement System",
        "_alias": "measure",
        "_since": "28",
        "metric": {
          "_deprecated": "false",
          "_description": "Metric System",
          "_since": "28"
        },
        "uksystem": {
          "_deprecated": "false",
          "_description": "UK System of measurement: feet, pints, etc.; pints are 20oz",
          "_alias": "imperial",
          "_since": "28"
        },
        "ussystem": {
          "_deprecated": "false",
          "_description": "US System of measurement: feet, pints, etc.; pints are 16oz",
          "_since": "28"
        }
      }
    }
  }
}
```

**transform-ime.json**
```json
{
  "keyword": {
    "t": {
      "i0": {
        "_deprecated": "false",
        "_description": "Input Method Engine transform:   Used to indicate an input method transformation, such as one used by   a client-side input method. The first subfield in a sequence would   typically be a 'platform' or vendor designation.",
        "_since": "21.0.2",
        "handwrit": {
          "_deprecated": "false",
          "_description": "Handwriting input: Used when the only information known (or requested) is that the text was (or is to be) converted using an handwriting input.",
          "_since": "21.0.2"
        },
        "pinyin": {
          "_deprecated": "false",
          "_description": "Pinyin input: for simplified Chinese characters. See also http://en.wikipedia.org/wiki/Pinyin_method.",
          "_since": "21.0.2"
        },
        "und": {
          "_deprecated": "false",
          "_description": "The choice of input method is not specified. Used when the only information known (or requested) is that the text was (or is to be) converted using an input method engine",
          "_since": "21.0.2"
        },
        "wubi": {
          "_deprecated": "false",
          "_description": "Wubi input: for simplified Chinese characters. For background information, see http://en.wikipedia.org/wiki/Wubi_method",
          "_since": "21.0.2"
        }
      }
    }
  }
}
```

File: tools/cldr-code/src/main/java/org/unicode/cldr/json/LdmlConvertRules.java
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/json/LdmlConvertRules.java:88]: oops, probably has no effect… i'll remove this

File: tools/cldr-code/src/main/java/org/unicode/cldr/json/LdmlConvertRules.java
Comment (srl295) [inline tools/cldr-code/src/main/java/org/unicode/cldr/json/LdmlConvertRules.java:167]: delete this comment

**Comment by nordzilla** [comment]
@srl295 

I think this is a definite improvement.

Using timezones as an example:

**Before**
```json
{
  "keyword": {
    "key": [
      {
        "type": {
          "_deprecated": "false",
          "_name": "adalv",
          "_description": "Andorra",
          "_alias": "Europe/Andorra"
        }
      },
      {
        "type": {
          "_deprecated": "false",
          "_name": "aedxb",
          "_description": "Dubai, United Arab Emirates",
          "_alias": "Asia/Dubai"
        }
      }
   ]
}
```

**After**
```json
{
  "keyword": {
    "u": {
      "tz": {
        "_deprecated": "false",
        "_description": "Time zone key",
        "_alias": "timezone",
        "adalv": {
          "_deprecated": "false",
          "_description": "Andorra",
          "_alias": "Europe/Andorra"
        },
        "aedxb": {
          "_deprecated": "false",
          "_description": "Dubai, United Arab Emirates",
          "_alias": "Asia/Dubai"
        }
      }
   }
}
```

I'm not really sure what the `"u"`  is. Is that a normal convention?

The underscores (which are present in both versions) also seem strange to me. 

The XML doesn't present them this way:
https://github.com/unicode-org/cldr/blob/master/common/bcp47/timezone.xml

Does it seem achievable to remove the underscores from all of these fields? e.g.

```json
{
  "keyword": {
    "u": {
      "tz": {
        "deprecated": "false",
        "description": "Time zone key",
        "alias": "timezone",
        "adalv": {
          "deprecated": "false",
          "description": "Andorra",
          "alias": "Europe/Andorra"
        },
        "aedxb": {
          "deprecated": "false",
          "description": "Dubai, United Arab Emirates",
          "alias": "Asia/Dubai"
        }
      }
   }
}
```

Or are the underscores desired?

**Comment by srl295** [comment]
The -u is actually a part of the locale code… So,

fr-CA-u-tz-adalv

The _ is a convention here, distinguishing an attribute from other values, that way there can be a sibling to tz for example￼￼￼

**Comment by srl295** [comment]
In your proposal with removing underscores, alias and adalv are siblings, and it's not clear programmatically that `alias` is not one of the time zone values￼￼

**Comment by srl295** [comment]
I'll add a regex for omitting the underscores. Although there could be a collision still, so leaf node only probably best.

**Review by sffc — APPROVED** [review]
LGTM since @nordzilla likes it, modulo a couple small suggestions

**Comment by srl295** [comment]
Minor tweak to `_deprecated`:
- omit if false
- output as boolean instead of string

**collation.json**

```json
{
  "keyword": {
    "u": {
      "co": {
        "_description": "Collation type key",
        "_alias": "collation",
        "big5han": {
          "_description": "Pinyin ordering for Latin, big5 charset ordering for CJK characters (used in Chinese)"
        },
        "compat": {
          "_description": "A previous version of the ordering, for compatibility",
          "_since": "26"
        },
        "dict": {
          "_description": "Dictionary style ordering (such as in Sinhala)",
          "_alias": "dictionary"
        },
        "direct": {
          "_deprecated": true,
          "_description": "Binary code point order (used in Hindi)"
        },
        "ducet": {
          "_description": "The default Unicode collation element table order",
          "_since": "2.0.1"
        },
        "emoji": {
          "_description": "Recommended ordering for emoji characters",
          "_since": "27"
        },
        "eor": {
          "_description": "European ordering rules",
          "_since": "24"
        },
        "gb2312": {
          "_description": "Pinyin ordering for Latin, gb2312han charset ordering for CJK characters (used in Chinese)",
          "_alias": "gb2312han"
        },
        "phonebk": {
          "_description": "Phonebook style ordering (such as in German)",
          "_alias": "phonebook"
        },
        "phonetic": {
          "_description": "Phonetic ordering (sorting based on pronunciation)"
        },
        "pinyin": {
          "_description": "Pinyin ordering for Latin and for CJK characters (used in Chinese)"
        },
        "reformed": {
          "_description": "Reformed ordering (such as in Swedish)"
        },
        "search": {
          "_description": "Special collation type for string search",
          "_since": "1.9"
        },
        "searchjl": {
          "_description": "Special collation type for Korean initial consonant search",
          "_since": "2.0.1"
        },
        "standard": {
          "_description": "Default ordering for each language"
        },
        "stroke": {
          "_description": "Pinyin ordering for Latin, stroke order for CJK characters (used in Chinese)"
        },
        "trad": {
          "_description": "Traditional style ordering (such as in Spanish)",
          "_alias": "traditional"
        },
        "unihan": {
          "_description": "Pinyin ordering for Latin, Unihan radical-stroke ordering for CJK characters (used in Chinese)"
        },
        "zhuyin": {
          "_description": "Pinyin ordering for Latin, zhuyin order for Bopomofo and CJK characters (used in Chinese)",
          "_since": "22"
        }
      },
      "ka": {
        "_description": "Collation parameter key for alternate handling",
        "_alias": "colAlternate",
        "noignore": {
          "_description": "Variable collation elements are not reset to ignorable",
          "_alias": "non-ignorable"
        },
        "shifted": {
          "_description": "Variable collation elements are reset to zero at levels one through three"
        }
      },
      "kb": {
        "_description": "Collation parameter key for backward collation weight",
        "_alias": "colBackwards",
        "false": {
          "_description": "No backwards (the second level to be forwards)",
          "_alias": "no"
        },
        "true": {
          "_description": "The second level to be backwards",
          "_alias": "yes"
        }
      },
      "kc": {
        "_description": "Collation parameter key for case level",
        "_alias": "colCaseLevel",
        "false": {
          "_description": "No special case level handling",
          "_alias": "no"
        },
        "true": {
          "_description": "The case level is inserted in front of tertiary",
          "_alias": "yes"
        }
      },
      "kf": {
        "_description": "Collation parameter key for ordering by case",
        "_alias": "colCaseFirst",
        "false": {
          "_description": "No special case ordering",
          "_alias": "no"
        },
        "lower": {
          "_description": "Lower case to be sorted before upper case"
        },
        "upper": {
          "_description": "Upper case to be sorted before lower case"
        }
      },
      "kh": {
        "_deprecated": true,
        "_description": "Collation parameter key for special Hiragana handling",
        "_alias": "colHiraganaQuaternary",
        "false": {
          "_description": "No special handling for Hiragana",
          "_alias": "no"
        },
        "true": {
          "_description": "Hiragana to be sorted before all non-variable on quaternary level",
          "_alias": "yes"
        }
      },
      "kk": {
        "_description": "Collation parameter key for normalization",
        "_alias": "colNormalization",
        "false": {
          "_description": "Skip normalization",
          "_alias": "no"
        },
        "true": {
          "_description": "Convert text into Normalization Form D before calculating collation weights",
          "_alias": "yes"
        }
      },
      "kn": {
        "_description": "Collation parameter key for numeric handling",
        "_alias": "colNumeric",
        "false": {
          "_description": "No special handling for numeric ordering",
          "_alias": "no"
        },
        "true": {
          "_description": "A sequence of decimal digits is sorted at primary level with its numeric value",
          "_alias": "yes"
        }
      },
      "kr": {
        "_valueType": "multiple",
        "_description": "Collation reorder codes",
        "_alias": "colReorder",
        "_since": "21",
        "currency": {
          "_description": "Currency reordering code, see LDML Part 5: Collation",
          "_since": "21"
        },
        "digit": {
          "_description": "Digit (number) reordering code, see LDML Part 5: Collation",
          "_since": "21"
        },
        "punct": {
          "_description": "Punctuation reordering code, see LDML Part 5: Collation",
          "_since": "21"
        },
        "REORDER_CODE": {
          "_description": "Other collation reorder code — for script, see LDML Part 5: Collation",
          "_since": "21"
        },
        "space": {
          "_description": "Whitespace reordering code, see LDML Part 5: Collation",
          "_since": "21"
        },
        "symbol": {
          "_description": "Symbol reordering code (other than currency), see LDML Part 5: Collation",
          "_since": "21"
        }
      },
      "ks": {
        "_description": "Collation parameter key for collation strength",
        "_alias": "colStrength",
        "identic": {
          "_description": "The identical level",
          "_alias": "identical"
        },
        "level1": {
          "_description": "The primary level",
          "_alias": "primary"
        },
        "level2": {
          "_description": "The secondary level",
          "_alias": "secondary"
        },
        "level3": {
          "_description": "The tertiary level",
          "_alias": "tertiary"
        },
        "level4": {
          "_description": "The quaternary level",
          "_alias": "quaternary quarternary"
        }
      },
      "kv": {
        "_description": "Collation parameter key for maxVariable, the last reordering group to be affected by ka-shifted",
        "_since": "25",
        "currency": {
          "_description": "Spaces, punctuation and all symbols are affected by ka-shifted",
          "_since": "25"
        },
        "punct": {
          "_description": "Spaces and punctuation are affected by ka-shifted (CLDR default)",
          "_since": "25"
        },
        "space": {
          "_description": "Only spaces are affected by ka-shifted",
          "_since": "25"
        },
        "symbol": {
          "_description": "Spaces, punctuation and symbols except for currency symbols are affected by ka-shifted (UCA default)",
          "_since": "25"
        }
      },
      "vt": {
        "_deprecated": true,
        "_valueType": "multiple",
        "_description": "Collation parameter key for variable top",
        "_alias": "variableTop",
        "CODEPOINTS": {
          "_description": "The variable top (one or more Unicode code points: LDML Appendix Q)"
        }
      }
    }
  }
}
```

**Comment by srl295** [comment]
Had to add some 'machinery' to propagate the original path… otherwise it got lost due to remapping.

**Comment by nordzilla** [comment]
This looks good to me as of my last build. 

I'm still not getting the removed underscores on leaves, but it's unclear to me if you intended to implement that or not. 

It seems like it's also find to keep it.

**Comment by srl295** [comment]
I was going to keep it consistent and keep the underscores.

**Comment by srl295** [comment]
@sffc if i can get a lgtm we can merge this

**Review by sffc — APPROVED** [review]

**Pull request merged**
